perun-rpc/src/main/java/cz/metacentrum/perun/rpc/methods/AuthzResolverMethod.java

```diff
@@ -129,6 +129,7 @@ public AuthzRoles call(ApiCaller ac, Deserializer parms) throws PerunException {
 			try {
 				bean = (PerunBean) Class.forName("cz.metacentrum.perun.core.api." + complementaryObjectName).getConstructor().newInstance();
 				bean.setId(complementaryObjectId);
+				bean = ac.fetchPerunBean(bean);
```

Comment:
Can't we simply fetch bean based on "class" and "id? If you would update ``getBeanFromDatabase()`` for that, you could use shared logic, instead of instantiating fake class and then replacing it with the real object anyway.

Reply:
You are right I will change it.

Reply:
Ok, we have decided that we will do it in the future as a refactorization.